### PR TITLE
Replace nominateNodeName annotation with PodStatus.NominatedNodeName

### PR DESCRIPTION
**Pull request by bsalamat**
**What this PR does / why we need it**:
Replaces nominateNodeName annotation with PodStatus.NominatedNodeName in scheudler's logic. We don't expect any logic/behavior changes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

ref #57471

/sig scheduling
cc: @k82cn @aveshagarwal @resouer

### CODE REVIEW

File: pkg/scheduler/core/scheduling_queue_test.go [hunk 1]
```diff
@@ -217,9 +221,12 @@ func TestUnschedulablePodsMap(t *testing.T) {
 				Name:      "p0",
 				Namespace: "ns1",
 				Annotations: map[string]string{
-					NominatedNodeAnnotationKey: "node1", "annot2": "val2",
+					"annot1": "val1",
```
Comment (aveshagarwal) [inline pkg/scheduler/core/scheduling_queue_test.go]: just noticed if its a typo? `annot2 -> annot1` and `val2->val1` 
@bsalamat can you double check, though tests are passing.

Comment (bsalamat) [reply pkg/scheduler/core/scheduling_queue_test.go]: No, it is not. That second annotation has no effect on the tests. I had put it there to have more annotations than just nominated node name. Now that the nominated node name is removed, I renamed the other annotation.

Comment (aveshagarwal) [reply pkg/scheduler/core/scheduling_queue_test.go]: sounds good.

**Pull request merged**